cmd/admin-bucket-remote-set.go

```diff
@@ -0,0 +1,214 @@
+	}
+	if argsNr > 2 {
+		fatalIf(errInvalidArgument().Trace(ctx.Args().Tail()...),
+			"Incorrect number of arguments for remote set command.")
+	}
+	if !ctx.IsSet("type") {
+		fatalIf(errInvalidArgument().Trace(ctx.Args()...), "type flag is required")
+	}
+
+}
+
+// RemoteMessage container for content message structure
+type RemoteMessage struct {
+	op           string
+	Status       string `json:"status"`
+	AccessKey    string `json:"accessKey,omitempty"`
+	SecretKey    string `json:"secretKey,omitempty"`
+	SourceBucket string `json:"sourceBucket"`
+	TargetURL    string `json:"TargetURL,omitempty"`
+	TargetBucket string `json:"TargetBucket,omitempty"`
+	RemoteARN    string `json:"RemoteARN,omitempty"`
+	Path         string `json:"path,omitempty"`
+	API          string `json:"api,omitempty"`
+	Type         string `json:"type"`
+}
+
+func (r RemoteMessage) String() string {
+	switch r.op {
+	case "list":
+		message := console.Colorize("TargetURL", fmt.Sprintf("%s ", r.TargetURL))
+		message += console.Colorize("SourceBucket", r.SourceBucket)
+		message += console.Colorize("Arrow", "->")
+		message += console.Colorize("TargetBucket", r.TargetBucket)
+		message += " "
+		message += console.Colorize("ARN", r.RemoteARN)
+		return message
+	case "remove":
+		return console.Colorize("RemoteMessage", "Removed Remote target for `"+r.SourceBucket+"` bucket successfully.")
+	case "set":
+		return console.Colorize("RemoteMessage", "Remote ARN = `"+r.RemoteARN+"`.")
+	}
+	return ""
+}
+
+// JSON returns jsonified message
+func (r RemoteMessage) JSON() string {
+	r.Status = "success"
+	jsonMessageBytes, e := json.MarshalIndent(r, "", " ")
+	fatalIf(probe.NewError(e), "Unable to marshal into JSON.")
+
+	return string(jsonMessageBytes)
+}
+
+// fetchRemoteTarget - returns the dest bucket, dest endpoint, access and secret key
+func fetchRemoteTarget(cli *cli.Context) (sourceBucket string, bktTarget *madmin.BucketTarget, err error) {
+	args := cli.Args()
+	argCount := len(args)
+	if argCount < 2 {
+		return sourceBucket, bktTarget, fmt.Errorf("Missing Remote target configuration")
+	}
+	_, sourceBucket = url2Alias(args[0])
+	TargetURL := args[1]
+	api := cli.String("api")
+	if api != "" && !isValidAPI(api) { // Empty value set to default "S3v4".
+		fatalIf(errInvalidArgument().Trace(api),
+			"Unrecognized API signature. Valid options are `[S3v4, S3v2]`.")
+	}
+	path := cli.String("path-style")
+	if !isValidPath(path) {
+		fatalIf(errInvalidArgument().Trace(path),
+			"Unrecognized bucket path style. Valid options are `[on,off, auto]`.")
+	}
+	u, cerr := url.Parse(TargetURL)
+	if cerr != nil {
+		fatalIf(probe.NewError(cerr), "Malformed Remote target URL")
+	}
+	arnType := cli.String("type")
+	if !madmin.ArnType(arnType).IsValid() {
+		fatalIf(errInvalidArgument().Trace(arnType), "Invalid --type flag. Valid option is `[replica]`.")
+	}
+	secure := u.Scheme == "https"
+	accessKey := u.User.Username()
+	secretKey, _ := u.User.Password()
```

Comment:
Use the regex used for MC_HOST_ instead - it is more portable rather that using url.Parse() @poornas